tests/test_linalg.py

```diff
@@ -210,25 +210,139 @@ def test_cholesky_errors(self):
+        ids=["None", "-dpnp.Inf", "-2", "-1", "1", "2", "dpnp.Inf", "fro"],
+    )
+    def test_cond_empty(self, shape, p):
+        a = numpy.empty(shape)
+        ia = inp.array(a)
+
+        result = inp.linalg.cond(ia, p=p)
+        expected = numpy.linalg.cond(a, p=p)
+        assert_dtype_allclose(result, expected)
+
+    @pytest.mark.parametrize(
+        "dtype", get_all_dtypes(no_bool=True, no_complex=True)
+    )
+    @pytest.mark.parametrize(
+        "shape", [(4, 4), (2, 4, 3, 3)], ids=["(4, 4)", "(2, 4, 3, 3)"]
+    )
+    @pytest.mark.parametrize(
+        "p",
+        [None, -inp.Inf, -2, -1, 1, 2, inp.Inf, "fro"],
+        ids=["None", "-dpnp.Inf", "-2", "-1", "1", "2", "dpnp.Inf", "fro"],
+    )
+    def test_cond(self, dtype, shape, p):
+        a = numpy.array(
+            numpy.random.uniform(-5, 5, numpy.prod(shape)), dtype=dtype
+        ).reshape(shape)
+        ia = inp.array(a)
+
+        result = inp.linalg.cond(ia, p=p)
+        expected = numpy.linalg.cond(a, p=p)
+        assert_dtype_allclose(result, expected)
+
+    @pytest.mark.parametrize(
+        "p",
+        [None, -inp.Inf, -2, -1, 1, 2, inp.Inf, "fro"],
+        ids=["None", "-dpnp.Inf", "-2", "-1", "1", "2", "dpnp.Inf", "fro"],
+    )
+    def test_cond_bool(self, p):
+        a = numpy.array([[True, True], [True, False]])
+        ia = inp.array(a)
+
+        result = inp.linalg.cond(ia, p=p)
+        expected = numpy.linalg.cond(a, p=p)
+        assert_dtype_allclose(result, expected)
+
+    @pytest.mark.parametrize("dtype", get_complex_dtypes())
+    @pytest.mark.parametrize(
+        "shape", [(4, 4), (2, 4, 3, 3)], ids=["(4, 4)", "(2, 4, 3, 3)"]
+    )
+    @pytest.mark.parametrize(
+        "p",
+        [None, -inp.Inf, -2, -1, 1, 2, inp.Inf, "fro"],
+        ids=["None", "-dpnp.Inf", "-2", "-1", "1", "2", "dpnp.Inf", "fro"],
+    )
+    def test_cond_complex(self, dtype, shape, p):
+        x1 = numpy.random.uniform(-5, 5, numpy.prod(shape))
+        x2 = numpy.random.uniform(-5, 5, numpy.prod(shape))
+        a = numpy.array(x1 + 1j * x2, dtype=dtype).reshape(shape)
+        ia = inp.array(a)
+
+        result = inp.linalg.cond(ia, p=p)
+        expected = numpy.linalg.cond(a, p=p)
+        assert_dtype_allclose(result, expected)
+
+    @pytest.mark.parametrize(
+        "p",
+        [-inp.Inf, -1, 1, inp.Inf, "fro"],
+        ids=["-dpnp.Inf", "-1", "1", "dpnp.Inf", "fro"],
+    )
+    def test_cond_nan_input(self, p):
+        a = numpy.array(numpy.random.uniform(-10, 10, 9)).reshape(3, 3)
+        a[1, 1] = numpy.nan
+        ia = inp.array(a)
+
+        result = inp.linalg.cond(ia, p=p)
+        expected = numpy.linalg.cond(a, p=p)
+        assert_dtype_allclose(result, expected)
+
+    @pytest.mark.parametrize(
+        "p",
+        [None, -inp.Inf, -2, -1, 1, 2, inp.Inf, "fro"],
+        ids=["None", "-dpnp.Inf", "-2", "-1", "1", "2", "dpnp.Inf", "fro"],
+    )
+    def test_cond_nan(self, p):
```

Comment:
the tests failed 4/5 times in coverage run, might be relating to using DPC++ RT 2023.2